src/api/identity/entity/requests.rs

```diff
@@ -0,0 +1,255 @@
+use rustify_derive::Endpoint;
+use serde::Serialize;
+use std::{collections::HashMap, fmt::Debug};
+
+use super::responses::{
+    CreateEntityResponse, ListEntitiesByIdResponse, ListEntitiesByNameResponse,
+    ReadEntityByIdResponse, ReadEntityByNameResponse,
+};
+
+/// ## Create an entity
+///
+/// This endpoint creates or updates an Entity.
+///
+/// * Path: identity/entity
+/// * Method: POST
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/entity#create-an-entity
+#[derive(Builder, Debug, Default, Endpoint)]
+#[endpoint(
+    path = "identity/entity",
+    method = "POST",
+    builder = "true",
+    response = "CreateEntityResponse"
+)]
+#[builder(setter(into, strip_option), default)]
+pub struct CreateEntityRequest {
+    /// Name of the entity.
+    pub name: Option<String>,
+    /// ID of the entity. If set, updates the corresponding existing entity.
+    pub id: Option<String>,
+    /// Metadata to be associated with the entity.
+    pub metadata: Option<HashMap<String, String>>,
+    /// Policies to be tied to the entity.
+    pub policies: Option<Vec<String>>,
+    /// Whether the entity is disabled. Disabled entities' associated tokens cannot be used, but are not revoked.
+    pub disabled: Option<bool>,
+}
+
+/// ## Read entity by ID
+///
+/// This endpoint queries the entity by its identifier.
+///
+/// * Path: identity/entity/id/{self.id}
+/// * Method: GET
+/// * Response: [ReadEntityByIdResponse]
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/entity#read-entity-by-id
+#[derive(Builder, Debug, Endpoint)]
+#[endpoint(
+    path = "identity/entity/id/{self.id}",
+    method = "GET",
+    builder = "true",
+    response = "ReadEntityByIdResponse"
+)]
+#[builder(setter(into))]
+pub struct ReadEntityByIdRequest {
+    /// Identifier of the entity.
+    #[endpoint(skip)]
+    pub id: String,
+}
+
+/// ## Update entity by ID
+///
+/// This endpoint is used to update an existing entity.
+///
+/// * Path: identity/entity/id/{self.id}
+/// * Method: POST
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/entity#update-entity-by-id
+#[derive(Builder, Debug, Default, Endpoint)]
+#[endpoint(
+    path = "identity/entity/id/{self.id}",
+    method = "POST",
+    builder = "true"
+)]
+#[builder(setter(into, strip_option), default)]
+pub struct UpdateEntityByIdRequest {
+    /// Identifier of the entity.
+    #[endpoint(skip)]
+    pub id: String,
+    /// Name of the entity.
+    pub name: Option<String>,
+    /// Metadata to be associated with the entity.
+    pub metadata: Option<HashMap<String, String>>,
+    /// Policies to be tied to the entity.
+    pub policies: Option<Vec<String>>,
+    /// Whether the entity is disabled. Disabled entities' associated tokens cannot be used, but are not revoked.
+    pub disables: Option<bool>,
+}
+
+/// ## Delete entity by ID
```

Comment:
We should use the TryFrom trait here to convert an `CreateEntityRequest` into an `UpdateEntityByIdRequest`
```suggestion
}

impl TryFrom<CreateEntityRequest> for UpdateEntityByIdRequest {
    type Error = &'static str;

    fn try_from(value: CreateEntityRequest) -> Result<Self, Self::Error> {
        match value.id {
            Some(id) => Ok(Self { id, name: value.name, metadata: value.metadata, policies: value.policies, disabled: value.disabled }),
            None => Err("id attribute has to be some"),
        }
    }
}

/// ## Delete entity by ID
```